File: clojure/basic.2.clj

(+ 1 1) 

; local variable
(let [x 3 y 4]
   (+ x y))

(let [[x y] [3 4]]
   (+ x y))

(let [x 3 y (* x x)]
   (+ x y))

; global variable
(def x 3)

; remove variable
(ns-unmap *ns* 'x)

nil

(nil? 3)

'x
(quote x)
(symbol? 'x)
(= 'x 'x)
:foo

(or (not true) (and true false))
(min 1 2 3)
(max 1 2 3)

(quot 7 3)
(rem 7 3)
(/ 7 3)
(/ 7 (* 3 1.0))

File: php/basic_2.php

<?php
// variable
{
	$v = NULL;
	$a = [];
	$d = [];
	$x = 1;
	list($y, $z) = [2, 3];
}
{
	list($g1, $g2) = [7, 8];

	function swap_globals() {
		global $g1, $g2;
		list($g1, $g2) = [$g2, $g1];
	}
}
{
	define("PI", 3.14);
	/// const PI = 3.14;
}
{
	list($x, $y, $z) = [1 ,2, 3];
	// 3 is discarded:
	list($x, $y) = [1, 2, 3];
	// z set to null
	/// list($x, $y, $z) = [1, 2];
}
{
	$x = 1;
	$y = ++$x;
	$z = --$y;
	$x > 0 ? $x : -$x;
}
{
	min(1, 2, 3);
	max(1, 2, 3);
	$a = [1, 2, 3];
	min($a);
	max($a);
}
?>
